pkg/client/split.go

```diff
@@ -75,21 +89,37 @@ func (d *delegatingClient) RESTMapper() meta.RESTMapper {
 type delegatingReader struct {
 	CacheReader  Reader
 	ClientReader Reader
+
+	uncachedGVKs map[schema.GroupVersionKind]struct{}
+	scheme       *runtime.Scheme
+}
+
+func (d *delegatingReader) isUncached(obj runtime.Object) (bool, error) {
+	gvk, err := apiutil.GVKForObject(obj, d.scheme)
+	if err != nil {
+		return false, err
+	}
+	_, isUncached := d.uncachedGVKs[gvk]
+	_, isUnstructured := obj.(*unstructured.Unstructured)
+	_, isUnstructuredList := obj.(*unstructured.UnstructuredList)
+	return isUncached || isUnstructured || isUnstructuredList, nil
```

Comment:
I would probably limit this strictly to checking if it's uncached, and keep the Unstructured check in Get and the UnstructuredList check in List.

Reply:
What I was wondering is if it would make sense to re-use the `isUncached` mechanism for this. The main issue is that ppl that set types as uncached probably do not want to overwrite the existing logic for unstructured with that.

Reply:
I moved it all here because unstructured and list are all uncached by default, although if that's preferable I can move it back

Reply:
So the current approach means that we allow configuring this except for `unstructured`. IMHO if we want to make this configurable, then we should also allow to disable it for `unstructured`, the current hardcoded assumption that someone who uses this type doesn't want it to get cached might or might not not be true (I've had cases where it wasn't). That in turns means that we need to split the configuration for this into a `SetUncached` and `AddUncached` (its completely fine if you only implement the latter here, but I don't want another interface change down the line, `WithUncached` is neither of the two).

This also has the advantage that we can better document the behavior on the constructors, right now there is no godoc mentioning the unstructured behavior